Cargo install fails with 'type annotations needed'

I wanted to try the language but running cargo install on a recent (`rustc 1.95.0 (59807616e 2026-04-14)`) version of Rust fails to build with a bunch of type inference errors. There's several but they all look similar to this. I tried building lisette `0.2.11` and `0.2.10` and both failed similarly:
```
error[E0283]: type annotations needed
   --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/lisette-semantics-0.2.11/src/passes/checks/generics.rs:157:63
    |
157 |                     .filter(|g| g.name.as_ref() == param_name.as_ref())
    |                                                               ^^^^^^
    |
    = note: multiple `impl`s satisfying `EcoString: AsRef<_>` found in the `ecow` crate:
            - impl AsRef<OsStr> for EcoString;
            - impl AsRef<Path> for EcoString;
            - impl AsRef<[u8]> for EcoString;
            - impl AsRef<str> for EcoString;
help: try using a fully qualified path to specify the expected types
    |
157 -                     .filter(|g| g.name.as_ref() == param_name.as_ref())
157 +                     .filter(|g| g.name.as_ref() == <EcoString as AsRef<T>>::as_ref(param_name))
    |

```

--- a/crates/emit/src/definitions/functions.rs
+++ b/crates/emit/src/definitions/functions.rs
@@ -477,7 +477,7 @@ impl Emitter<'_> {
             if param.ty.is_ref()
                 && let Some(inner) = param.ty.inner()
                 && let Type::Parameter(name) = &inner
-                && bounded_generics.contains(name.as_ref())
+                && bounded_generics.contains(name.as_str())
             {
                 self.function_state
                     .record_absorbed_ref_generic(name.to_string());

--- a/crates/semantics/src/passes/checks/generics.rs
+++ b/crates/semantics/src/passes/checks/generics.rs
@@ -126,11 +126,11 @@ fn check_constrained_return_type(
     let span = return_annotation.get_span();
     for return_param in params.iter() {
         if let Type::Parameter(param_name) = return_param
-            && let Some(method_bounds) = required_bounds.get(param_name.as_ref())
+            && let Some(method_bounds) = required_bounds.get(param_name.as_str())
         {
             let fn_generic = generics
                 .iter()
-                .find(|g| g.name.as_ref() == param_name.as_ref());
+                .find(|g| g.name.as_str() == param_name.as_str());
 
             if let Some(fn_gen) = fn_generic {
                 let fn_bounds: Vec<Type> = fn_gen
@@ -154,7 +154,7 @@ fn check_constrained_return_type(
             } else if let Some(impl_generics) = enclosing_impl_generics {
                 let impl_bounds: Vec<Type> = impl_generics
                     .iter()
-                    .filter(|g| g.name.as_ref() == param_name.as_ref())
+                    .filter(|g| g.name.as_str() == param_name.as_str())
                     .flat_map(|g| g.bounds.iter())
                     .filter_map(|b| annotation_to_type(b, module_id, store))
                     .collect();

